cli/cmd/add/component-version/cmd.go

```diff
@@ -224,15 +226,39 @@ func GetRepositorySpec(cmd *cobra.Command) (runtime.Typed, error) {
 	if err != nil {
 		return nil, fmt.Errorf("getting repository reference flag failed: %w", err)
 	}
-	var accessMode ctfv1.AccessMode = ctfv1.AccessModeReadWrite
-	if !repoRef.Exists() {
-		accessMode += "|" + ctfv1.AccessModeCreate
+
+	// Get the repository type.
+	repoTypeString, err := compref.GuessType(repoRef.String())
```

Comment:
I really dont like that we use a repoRef from the file flag here. IMO the file flag should be replaced by a component ref parsing which already includes guessing the type.